torch/_functorch/partitioners.py

```diff
@@ -1449,14 +1450,28 @@ def materialize_arg(x):
             shape = list(x.meta["val"].shape)
 
             def realize_symbol(d):
-                return hint_int(d, fallback=4096)
+                if isinstance(d, torch.SymInt) and d.node.hint is None:
+                    unbacked_symbols = free_unbacked_symbols(d.node.expr)
+                    replacements = {
+                        s: 4096
+                        if s in unbacked_symbols
+                        else d.node.shape_env.var_to_val[s]
+                        for s in d.node.expr.free_symbols
+                    }
+                    return d.node.expr.xreplace(replacements)
+                return hint_int(d)
 
             shape = [realize_symbol(s) for s in shape]
             return x.meta["val"].new_empty_strided(
                 shape, stride=x.meta["tensor_meta"].stride
             )
         elif isinstance(x, fx.Node) and isinstance(x.meta["val"], torch.SymInt):
-            return hint_int(x.meta["val"], fallback=4096)
+            d = x.meta["val"]
+            if d.node.hint is None:
```

Comment:
this code and the code above do the same things except here we replace all free symbols with 4096
even backed? and above we replace unbacked with 4096 and backed with their hints.

I wonder if:
1. do we also want want to replace backed with their hints here also?  would it make sense to pull the logic of realize_symbol in a function and call it from both places?

2. do you think it i s worth pushing the logic maybe inside hint_int(unbacked_repl = closure()..)
or do you think probably not many people do this stuff in such case not worth the trouble?

3. you could also potentially only replace unbacked in the expressions and then call 
 def symbolic_hint(self, expr: Union[Expr, int]) -> Union[Expr, int] on the result to replace backed.

Reply:
> do we also want want to replace backed with their hints here also?

hmm, isn't this change already properly replacing backed ints with their hints?
```
        unbacked_symbols = free_unbacked_symbols(d.node.expr)
        replacements = {
            s: 4096 if s in unbacked_symbols else d.node.shape_env.var_to_val[s]
            for s in d.node.expr.free_symbols
        }
        return d.node.expr.xreplace(replacements)
```
(I tried to take all the free symbols, and assume that every symbol returned by "free_unbacked_symbols()" have a hint that lives in the `var_to_val` dict)